### ISSUE TITLE
[libc++] <format> has uninitialized variables when formatting a string with zero arguments. 

### ISSUE THREAD

**Issue by BlamKiwi** [llvm/llvm-project#63421]
When formatting a string with 0 arguments,  `std::format("Hello World!");` , GCC 13.1 complains there are maybe uninitialized variables. This causes spurious warnings/errors with appropriate build flags enabled. 

```
[build] /workspaces/include/c++/v1/__format/format_functions.h:71:65: error: '<anonymous>' may be used uninitialized [-Werror=maybe-uninitialized]
[build]    71 |   return _VSTD::__format_arg_store<_Context, _Args...>(__args...);
```

The culprit is in the constructor of  `__format_arg_store`. When constructing an argument store of 0 size, it never writes a value to `__packed_format_arg_store::__types_`. A quick fix is to give `__packed_format_arg_store::__types_` a default value of 0.

```
template <class _Context, size_t N>
struct __packed_format_arg_store {
  __basic_format_arg_value<_Context> __values_[N];
  uint64_t __types_ = 0;
};
```

**Comment by mordante** [comment]
This is not a real bug, when the size is 0 the value will never be used. However I don't have an issue with fixing it too. I'll create a patch later this week.
